### DISCUSSION TITLE
Newbee question (Xbox controller logic)

### DISCUSSION THREAD

**Discussion by Carrera124** [pybricks/support#1748] [General]
Hi, I am doing my first steps using pybricks, and I started with the preconfigured projects for the Lego sets.

Most of the blocks and code are easy to understand.
But I have one question regarding the Xbox controller usage.
In all examples (42109, 42124, 42129) the speed control is calculated by "right trigger" minus "left trigger".
What's the reason to use a mathematical subtraction in this case?
If I would have approached it myself, my first try would have been to simply check which button is pressed?

**Comment by laurensvalk** [comment]
These triggers each give a gradual value from 0 to 100.

By doing (right - left), you get a result that ranges gradually between -100 for left fully pressed, to 0 for both released (or both pressed), to +100 for the right fully pressed.

This gives a reasonably intuitive and convenient control with very few blocks, and lets you drive forward and in reverse at variable speed. 

There's nothing wrong with building your own logic with if / else blocks, though! If you want to share your end result, I'm sure there's people here who would love to learn from it :)

**Comment by Carrera124** [comment]
Thank you :-) I am just exploring.
I started coding on a C64, with BASIC and Assembler.
Later moved on to PCs, with C, C++, C#, Java, etc.
But today's coding is somewhat different. Back then, we had to "invent" many things from scratch.
Today, there are predefined libraries and classes for nearly everything. So, my main task is to find out about the APIs and their possibilities.
